lib/fizzy/instructions.cpp

```diff
@@ -227,11 +227,237 @@ constexpr InstructionMetrics instruction_metrics_table[256] = {
     /* f32_reinterpret_i32 = 0xbe */ {1, 0},
     /* f64_reinterpret_i64 = 0xbf */ {1, 0},
 };
+
+// Order of input parameters is the order they are popped from stack,
+// which is consistent with the order in FuncType::inputs.
+constexpr InstructionType instruction_type_table[256] = {
+    // 5.4.1 Control instructions
+    /* unreachable         = 0x00 */ {{}, {}},
+    /* nop                 = 0x01 */ {{}, {}},
+    /* block               = 0x02 */ {{}, {}},
+    /* loop                = 0x03 */ {{}, {}},
+    /* if_                 = 0x04 */ {{ValType::i32}, {}},
+    /* else_               = 0x05 */ {{}, {}},
```

Comment:
Why have these initialisers `{{}, {}},` when empty opcodes can go with `{}` ?

Reply:
It can, I just found it somewhat more explicit to show both input and output empty. Would you prefer to make it `{}`?

Reply:
It doesn't really make any difference.